R/decisiontreeutils.R

```diff
@@ -0,0 +1,363 @@
+#' @param config list of config options
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+#' @return params to call AlteryxGraph on
+getDTGraphCalls <- function(config, the_model, is_XDF) {
+  # Address the user plot parameters and create the plots
+
+  # The values in the leaf summary
+  leaf_sum <- 4
+  if (config$do.counts == "True")
+    leaf_sum <- 2
+  if (the_model$method != "class")
+    leaf_sum <- 0
+  print(the_model$method)
+
+  # Uniform or proportional tree branch lengths
+  uniform <- "FALSE"
+  fallen <- "TRUE"
+  if (config$b.dist == "True") {
+    uniform <- "TRUE"
+    fallen <- "FALSE"
+  }
+
+  # assemble list of function and params to pass to AlteryxGraph
+  graph_calls <- list()
+
+  graph_calls$tree <- list()
+  graph_calls$tree$f <- "rpart.plot"
+  graph_calls$tree$args <- list (
+    the_model_rpart,
+    type = 0,
+    extra = leaf_sum,
+    uniform = uniform,
+    fallen.leaves = fallen,
+    main = "Tree Plot",
+    cex = 1
+  )
+
+  graph_calls$prune <- list(
+    f = "plotcp",
+    args = list(the_model)
+  )
+}
+
+#' get component for interactive viz
+#'
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+getDTViz <- function(the_model, is_XDF) {
+
+  ## Interactive Visualization
+  library(AlteryxRviz)
+  library(htmltools)
+  if (is_XDF){
+    k1 = tags$div(tags$h4(
+      "Interactive Visualizations are not supported for Revolution Enterprise"
+    ))
+    # renderInComposer(k1, nOutput = 5)
+  } else {
+    if (!(packageVersion('AlteryxRviz') >= "0.2.5")){
+      k1 = tags$div(
+        tags$h4("You need AlteryxRviz >= 0.2.5")
+      )
+    } else {
+      #the_model = rpart(Species ~ ., data = iris)
+      tooltipParams = list(
+        width = '250px',
+        top = '130px',
+        left = '100px'
+      )
+      dt = renderTree(the_model, tooltipParams = tooltipParams)
+      vimp = varImpPlot(the_model, height = 300)
+
+      cmat = if (!is.null(the_model$frame$yval2)){
+        iConfusionMatrix(getConfMatrix(the_model), height = 300)
+      }  else {
+        tags$div(h1('Confusion Matrix Not Valid'), height = 300)
+      }
+
+      k1 = dtDashboard(dt, vimp, cmat)
+    }
+    # renderInComposer(k1, nOutput = 5)
+  }
+  k1
+}
+
+#' get results in form of output list
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+getOutputsDT <- function(config, the_model, is_XDF) {
+  # Assemble list to return needed elements to output
+  results <- list()
+
+  results$output1 <- getDTPipes(config, the_model, is_XDF)
+  results$output3 <- modelOut(config$model.name, the_model)
+
+  graph_results <- getDTgraphCalls(config, the_model, is_XDF)
+  results$output2 <- graph_results$tree
+  results$output4 <- graph_results$prune
+
+  results$output5 <- getDTViz(the_model, is_XDF)
+
+  write.Alteryx(results$output1, nOutput = 1)
+  write.Alteryx(results$output3, nOutput = 3)
+
```

Comment:
Avoid Alteryx specific I/O functions as far as possible. This keeps the package testable in Travis. I think best practice should be to use the package to create the objects that will then get passed on to `write.Alteryx` and `AlteryxGraph`. Another advantage this gives us is that the tool author can then modify the results in the macro before outputting them, thereby not having to mess around with updating the package and building it. This will be useful especially for customers who want to tweak our macros.